### PR TITLE
LPD-31980 update old data in database with the new classname of the rawMetadataProcessor after moving the class to other package

### PR DESCRIPTION
**Pull request by AliciaGarciaGarcia**
LPD-31980 Upgrade fails on duplicate record in ddmStructure table

# What is this trying to solve?

https://liferay.atlassian.net/browse/LPD-31980

After the changes made on https://liferay.atlassian.net/browse/LPS-188559 that moved the package of the RawMetadataProcessor was a miss match between the name on the table classname and the RawMetadataProcessor classname

# How am I fixing it?

We are updating the value on the Classname_ table if needed

# How can you verify that it works?

Run the included automated tests.
Or check https://liferay.atlassian.net/browse/LPP-54654 with the client database

# Commits

- **LPD-31980 update old data in database with the new classname of the rawMetadataProcessor after moving the class to other package**
- **LPD-31980 create test to check if on upgrade the classname is changed if previously has old class package**

### CODE REVIEW

**Comment by AliciaGarciaGarcia** [comment]
ci:test:upgrade

**Comment by AliciaGarciaGarcia** [comment]
ci:test:help

**Comment by AliciaGarciaGarcia** [comment]
ci:test:upgrade-all

**Comment by AliciaGarciaGarcia** [comment]
ci:report:43618618

File: modules/apps/portal/portal-upgrade-test/src/testIntegration/java/com/liferay/portal/upgrade/v7_0_1/test/UpgradeDocumentLibraryTest.java [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+		}
+	}
+
+	@Test
+	public void testUpgrade() throws Exception {
+		long classNameId = RandomTestUtil.randomLong();
+
+		Connection connection = DataAccess.getConnection();
+
+		long originalClassNameId = _getClassNameIdFromClassName(
```
Comment (jkappler) [inline modules/apps/portal/portal-upgrade-test/src/testIntegration/java/com/liferay/portal/upgrade/v7_0_1/test/UpgradeDocumentLibraryTest.java:87]: @AliciaGarciaGarcia, className will always exist in this version of the portal, so we can skip the checking

**Review by jkappler — CHANGES_REQUESTED** [review]

File: modules/apps/portal/portal-upgrade-test/src/testIntegration/java/com/liferay/portal/upgrade/v7_0_1/test/UpgradeDocumentLibraryTest.java (hunk 1 above)
Comment (AliciaGarciaGarcia) [reply modules/apps/portal/portal-upgrade-test/src/testIntegration/java/com/liferay/portal/upgrade/v7_0_1/test/UpgradeDocumentLibraryTest.java:87]: Thanks

I will remove the conditionals

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Review by jkappler — APPROVED** [review]

**Comment by jkappler** [comment]
ci:forward

**Comment by AliciaGarciaGarcia** [comment]
ci:report:44347850

**Comment by AliciaGarciaGarcia** [comment]
ci:forward:force

**Pull request closed without merging**